### ISSUE TITLE
allow to disable or repeat optimization passes

### ISSUE THREAD

**Issue by sigmundch** [dart-lang/sdk#25452]
It should be possible (correct for both the implementation and generated code) to disable or repeat individual optimization passes.  This will help find bugs and makes it easier to understand performance characteristics.

**Comment by srawlins** [comment]
@sigmundch: There's currently no flag to disable optimizations, correct?

CC @goderbauer @antigrav

**Comment by sigmundch** [comment]
correct, codegen expects certain passes to run first. Not all passes are needed, but there is no flags available today to disable the unnecessary passes.

**Comment by sigmundch** [comment]
a requirement we might add here: the backend should be smart about ignoring types that it discovers that resolution knows that are not needed. This can happen when some phases are disabled and we end up with worse type information than what global type inference had.
